pkg/controller/machine/machine.go

```diff
@@ -801,36 +799,32 @@ func (c *Controller) handleObject(obj interface{}) {
 		glog.V(4).Infof("Recovered deleted object '%s' from tombstone", object.GetName())
 	}
 
-	ownerRef := metav1.GetControllerOf(object)
-	if ownerRef != nil {
-		if ownerRef.Kind != "Machine" {
-			return
-		}
-		machine, err := c.machinesLister.Get(ownerRef.Name)
-		if err != nil {
-			glog.V(4).Infof("ignoring orphaned object '%s' of machine '%s'", object.GetSelfLink(), ownerRef.Name)
-			return
-		}
-
-		glog.V(6).Infof("Processing node: %s (machine=%s)", object.GetName(), machine.Name)
-		c.enqueueMachine(machine)
+	machinesList, err := c.machinesLister.List(labels.Everything())
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("Failed to list machines in lister: %v", err))
 		return
 	}
 
-	if ownerRef == nil {
-		machines, err := c.machinesLister.List(labels.Everything())
-		if err != nil {
-			utilruntime.HandleError(fmt.Errorf("error listing machines: '%v'", err))
-			return
-		}
-		for _, machine := range machines {
-			// We get triggered by node{Add,Update}, so enqeue machines if they
-			// have no nodeRef yet to make matching happen ASAP
+	var ownerUIDString string
+	var exists bool
+	if ownerUIDString, exists = object.GetLabels()[NodeOwnerLabelName]; !exists {
```

Comment:
Do we have a guarantee that this always returns an initialized map?
I remember having issues with the annotations map and objects from the lister